File: avalon/framework/src/impl/org/apache/avalon/framework/logger/ConsoleLogger.java
/*
 * Copyright (C) The Apache Software Foundation. All rights reserved.
 *
 * This software is published under the terms of the Apache Software License
 * version 1.1, a copy of which has been included with this distribution in
 * the LICENSE.txt file.
 */
package org.apache.avalon.framework.logger;


/**
 * Logger sending everything to the standard output streams.
 * This is mainly for the cases when you have a utility that
 * does not have a logger to supply.
 *
 * @author <a href="mailto:leo.sutic@inspireinfrastructure.com">Leo Sutic</a>
 * @author <a href="mailto:bloritsch@apache.org">Berin Loritsch</a>
 */
public final class ConsoleLogger implements Logger
{
    public final static int LEVEL_DEBUG = 0;
    public final static int LEVEL_INFO = 1;
    public final static int LEVEL_WARN = 2;
    public final static int LEVEL_ERROR = 3;
    public final static int LEVEL_FATAL = 4;
    public final static int LEVEL_DISABLED = 5;

    private final int m_logLevel;

    /**
     * Creates a new ConsoleLogger with the priority set to DEBUG.
     */
    public ConsoleLogger()
    {
        this( LEVEL_DEBUG );
    }

    /**
     * Creates a new ConsoleLogger.
     */
    public ConsoleLogger( final int logLevel )
    {
        m_logLevel = logLevel;
    }

    public void debug( final String message )
    {
        debug( message, null );
    }

    public void debug( final String message, final Throwable throwable )
    {
        if ( m_logLevel <= LEVEL_DEBUG )
        {
            System.out.print( "[DEBUG] " );
            System.out.println( message );

            if ( null != throwable )
            {
                throwable.printStackTrace( System.out );
            }
        }
    }

    public boolean isDebugEnabled()
    {
        return m_logLevel <= LEVEL_DEBUG;
    }

    public void info( final String message )
    {
        info( message, null );
    }

    public void info( final String message, final Throwable throwable )
    {
        if ( m_logLevel <= LEVEL_INFO )
        {
            System.out.print( "[INFO] " );
            System.out.println( message );

            if ( null != throwable )
            {
                throwable.printStackTrace( System.out );
            }
        }
    }

    public boolean isInfoEnabled()
    {
        return m_logLevel <= LEVEL_INFO;
    }

    public void warn( final String message )
    {
        warn( message, null );
    }

    public void warn(final String message, final Throwable throwable)
    {
        if ( m_logLevel <= LEVEL_WARN )
        {
            System.out.print( "[WARNING] " );
            System.out.println( message );

            if ( null != throwable )
            {
                throwable.printStackTrace( System.out );
            }
        }
    }

    public boolean isWarnEnabled()
    {
        return m_logLevel <= LEVEL_WARN;
    }

    public void error( final String message )
    {
        error( message, null );
    }

    public void error( final String message, final Throwable throwable )
    {
        if ( m_logLevel <= LEVEL_ERROR )
        {
            System.out.print( "[ERROR] " );
            System.out.println( message );

            if ( null != throwable )
            {
                throwable.printStackTrace( System.out );
            }
        }
    }

    public boolean isErrorEnabled()
    {
        return m_logLevel <= LEVEL_ERROR;
    }

    public void fatalError( final String message )
    {
        fatalError( message, null );
    }

    public void fatalError( final String message, final Throwable throwable )
    {
        if ( m_logLevel <= LEVEL_FATAL )
        {
            System.out.print( "[FATAL ERROR] " );
            System.out.println( message );

            if ( null != throwable )
            {
                throwable.printStackTrace( System.out );
            }
        }
    }

    public boolean isFatalErrorEnabled()
    {
        return m_logLevel <= LEVEL_FATAL;
    }

    public Logger getChildLogger( final String name )
    {
        return this;
    }
}

File: central/laboratory/avalon-net/csframework/AvalonFramework/Logger/ConsoleLogger.cs
// Copyright 2003-2004 The Apache Software Foundation
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.namespace Apache.Avalon.Framework{	using System;
	/// <summary>	///	The Logger sending everything to the standard output streams.	/// This is mainly for the cases when you have a utility that	/// does not have a logger to supply.	/// </summary>	public class ConsoleLogger : ILogger	{		/// <summary>
		/// Default logger level
		/// </summary>		private LoggerLevel m_logLevel = LoggerLevel.Debug;		/// <summary>
		/// Default name
		/// </summary>		private String m_name = String.Empty;		/// <summary>		/// Creates a new ConsoleLogger with the priority set to DEBUG.		/// </summary>		public ConsoleLogger(): this(LoggerLevel.Debug)		{		}		/// <summary>		/// Creates a new ConsoleLogger.		/// </summary>		/// <param name="logLevel">The Log level typecode.</param>		public ConsoleLogger(LoggerLevel logLevel)		{			this.m_logLevel = logLevel;		}		/// <summary>		/// Creates a new ConsoleLogger.		/// </summary>		/// <param name="name">The Log name.</param>		public ConsoleLogger(String name)		{			this.m_name = name;		}		/// <summary>		/// Creates a new ConsoleLogger.		/// </summary>		/// <param name="name">The Log name.</param>		/// <param name="logLevel">The Log level typecode.</param>		public ConsoleLogger(String name, LoggerLevel logLevel) : this(name)		{			this.m_logLevel = logLevel;		}		/// <summary>		/// Logs a debug message.		/// </summary>		/// <param name="message">The Message</param>		public void Debug(string message)		{			Debug(message, null as Exception);		}		/// <summary>		/// Logs a debug message. 		/// </summary>		/// <param name="message">The Message</param>		/// <param name="exception">The Exception</param>		public void Debug(string message, Exception exception)		{			Log(LoggerLevel.Debug, message, exception); 		}		/// <summary>		/// Logs a debug message.		/// </summary>		/// <param name="format">Message format</param>		/// <param name="args">Array of objects to write using format</param>		public void Debug( string format, params Object[] args )		{			Debug(String.Format(format, args));		}		/// <summary>		/// Determines if messages of priority "debug" will be logged.		/// </summary>		/// <value>True if "debug" messages will be logged.</value> 		public bool IsDebugEnabled		{			get			{				return (m_logLevel <= LoggerLevel.Debug);			}		}		/// <summary>		/// Logs an info message.		/// </summary>		/// <param name="message">The Message</param>		public void Info( string message )		{			Info(message, null as Exception);		}		/// <summary>		/// Logs an info message. 		/// </summary>		/// <param name="message">The Message</param>		/// <param name="exception">The Exception</param>		public void Info( string message, Exception exception)		{			Log(LoggerLevel.Info, message, exception); 		}		/// <summary>		/// Logs an info message.		/// </summary>		/// <param name="format">Message format</param>		/// <param name="args">Array of objects to write using format</param>		public void Info( string format, params Object[] args )		{			Info(String.Format(format, args));		}		/// <summary>		/// Determines if messages of priority "info" will be logged.		/// </summary>		/// <value>True if "info" messages will be logged.</value>		public bool IsInfoEnabled		{			get			{				return (m_logLevel <= LoggerLevel.Info);			}		}		/// <summary>		/// Logs a warn message.		/// </summary>		/// <param name="message">The Message</param>		public void Warn(string message )		{			Warn(message, null as Exception);		}		/// <summary>		/// Logs a warn message. 		/// </summary>		/// <param name="message">The Message</param>		/// <param name="exception">The Exception</param>		public void Warn(string message, Exception exception)		{			Log(LoggerLevel.Warn, message, exception); 		}		/// <summary>		/// Logs an warn message.		/// </summary>		/// <param name="format">Message format</param>		/// <param name="args">Array of objects to write using format</param>		public void Warn( string format, params Object[] args )		{			Warn(String.Format(format, args));		}		/// <summary>		/// Determines if messages of priority "warn" will be logged.		/// </summary>		/// <value>True if "warn" messages will be logged.</value>		public bool IsWarnEnabled		{			get			{				return (m_logLevel <= LoggerLevel.Warn);			}		}		/// <summary>		/// Logs an error message.		/// </summary>		/// <param name="message">The Message</param>		public void Error(string message )		{			Error(message, null as Exception);		}		/// <summary>		/// Logs an error message. 		/// </summary>		/// <param name="message">The Message</param>		/// <param name="exception">The Exception</param>		public void Error(string message, Exception exception)		{			Log(LoggerLevel.Error, message, exception); 		}
		/// <summary>		/// Logs an error message.		/// </summary>		/// <param name="format">Message format</param>		/// <param name="args">Array of objects to write using format</param>		public void Error( string format, params Object[] args )		{			Error(String.Format(format, args));		}		/// <summary>		/// Determines if messages of priority "error" will be logged.		/// </summary>		/// <value>True if "error" messages will be logged.</value>		public bool IsErrorEnabled		{			get			{				return (m_logLevel <= LoggerLevel.Error);			}		}		/// <summary>		/// Logs a fatal error message.		/// </summary>		/// <param name="message">The Message</param>		public void FatalError(string message )		{			FatalError(message, null as Exception);		}		/// <summary>		/// Logs a fatal error message.		/// </summary>		/// <param name="message">The Message</param>		/// <param name="exception">The Exception</param>		public void FatalError(string message, Exception exception)		{			Log(LoggerLevel.Fatal, message, exception); 		}		/// <summary>		/// Logs a fatal error message.		/// </summary>		/// <param name="format">Message format</param>		/// <param name="args">Array of objects to write using format</param>		public void FatalError( string format, params Object[] args )		{			FatalError(String.Format(format, args));		}		/// <summary>		/// Determines if messages of priority "fatalError" will be logged.		/// </summary>		/// <value>True if "fatalError" messages will be logged.</value>		public bool IsFatalErrorEnabled		{			get 			{				return (m_logLevel <= LoggerLevel.Fatal); 			}		}		/// <summary>		/// A Common method to log.		/// </summary>		/// <param name="level">The level of logging</param>		/// <param name="levelName">The Level name</param>		/// <param name="message">The Message</param>		/// <param name="exception">The Exception</param>		protected void Log(LoggerLevel level, string message, Exception exception) 		{			if(m_logLevel <= level)			{				Console.Out.WriteLine(string.Format("[{0}] '{1}' {2}", level.ToString(), m_name, message));								if(exception != null)				{					Console.Out.WriteLine(exception.StackTrace);				}			}		}		/// <summary>		///	Just returns this logger (<c>ConsoleLogger</c> is not hierarchical).		/// </summary>		/// <param name="name">Ignored</param>		/// <returns>This ILogger instance.</returns> 		public ILogger CreateChildLogger(string name )		{			return new ConsoleLogger( String.Format("{0}.{1}", this.m_name, name), m_logLevel );		}	}}